Makefile

```diff
@@ -9,9 +9,9 @@ BUILD_DATE             = $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 GIT_COMMIT             = $(shell git rev-parse HEAD)
 GIT_REMOTE             = origin
 GIT_BRANCH             = $(shell git rev-parse --symbolic-full-name --verify --quiet --abbrev-ref HEAD)
-GIT_TAG                = $(shell git describe --always --tags --abbrev=0 || echo untagged)
+GIT_TAG                = $(shell git describe --exact-match --abbrev=0  2> /dev/null || echo untagged)
 GIT_TREE_STATE         = $(shell if [ -z "`git status --porcelain`" ]; then echo "clean" ; else echo "dirty"; fi)
-DEV_BRANCH             = $(shell [ $(GIT_BRANCH) = master ] || [ `echo $(GIT_BRANCH) | cut -c -8` = release- ] && echo false || echo true)
+DEV_BRANCH             = $(shell [ $(GIT_BRANCH) = master ] || [ `echo $(GIT_BRANCH) | cut -c -8` = release- ] || [[ "$(GIT_TAG)" =~ ^v[0-9]+\.[0-9]+\.[0-9]+.*$$ ]] && echo false || echo true)
```

Comment:
`DEV_BRANCH` has been broken when we are releasing. We release on a tag (eg `v3.0.0-rc5`). When git is on a tag, it is on detached state which means that `GIT_BRANCH` will be `HEAD` and `DEV_BRANCH` will always be true.

In the past this didn't matter when building because the UI was built by a separate Docker step and copied into this current directory, so when the Makefile arrived at this spot and -- because of `DEV_BRANCH` being incorrect, `STATIC_FILES` here was actually incorrectly `false` -- nothing would happen because `ui/dist/app/index.html` already existed.

```
ifeq ($(STATIC_FILES),true)
ui/dist/app/index.html: $(shell find ui/src -type f && find ui -maxdepth 1 -type f)
	# `yarn install` is fast (~2s), so you can call it safely.
	JOBS=max yarn --cwd ui install
	# `yarn build` is slow, so we guard it with a up-to-date check.
	JOBS=max yarn --cwd ui build
else
ui/dist/app/index.html:
	@mkdir -p ui/dist/app
	echo "Built without static files" > ui/dist/app/index.html
endif
```

However, when we changed the makefile to this:

```
ui/dist/app/index.html: $(shell find ui/src -type f && find ui -maxdepth 1 -type f)
	# `yarn install` is fast (~2s), so you can call it safely.
	JOBS=max yarn --cwd ui install
	# `yarn build` is slow, so we guard it with a up-to-date check.
	JOBS=max yarn --cwd ui build

$(GOPATH)/bin/staticfiles:
	cd `mktemp -d` && go get bou.ke/staticfiles

ifeq ($(STATIC_FILES),true)
server/static/files.go: $(GOPATH)/bin/staticfiles ui/dist/app/index.html
	# Pack UI into a Go file.
	$(GOPATH)/bin/staticfiles -o server/static/files.go ui/dist/app
else
server/static/files.go:
	cp ./server/static/files.go.stub ./server/static/files.go
endif
```

The fact that `ui/dist/app/index.html` existed no longer protected the fact that `STATIC_FILES` was incorrect (because `DEV_BRANCH` was also incorrect).

Reply:
I trust you.